go-controller/pkg/ovn/gateway_init.go

```diff
@@ -59,9 +59,12 @@ func (oc *DefaultNetworkController) cleanupStalePodSNATs(nodeName string, nodeIP
 	podIPsOnNode := sets.NewString() // collects all podIPs on node
 	for _, pod := range pods.Items {
 		pod := pod
+		if !util.PodScheduled(&pod) || util.PodCompleted(&pod) {
```

Comment:
so if a pod is completed it will still have an IP on it (although not in use by networking anymore). Shouldn't we try to delete that NAT if it is not in use by any other pod? We have some similar logic that detects that: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/pod_selector_address_set.go#L444

Reply:
is it possible to have a perf impact if we are checking that ip against all other pod ips to make sure its not in use?